pipelined/bevy_ui2/src/render/render_pass.rs

```diff
@@ -0,0 +1,197 @@
+        vec![SlotInfo::new(UiPassNode::IN_VIEW, SlotType::Entity)]
+    }
+
+    fn update(&mut self, world: &mut World) {
+        self.query.update_archetypes(world);
+    }
+
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let view_entity = graph.get_input_entity(Self::IN_VIEW)?;
+        let (transparent_phase, target) = self
+            .query
+            .get_manual(world, view_entity)
+            .expect("view entity should exist");
+        let pass_descriptor = RenderPassDescriptor {
+            label: Some("ui_pass"),
+            color_attachments: &[RenderPassColorAttachment {
+                view: &target.view,
+                resolve_target: None,
+                ops: Operations {
+                    load: LoadOp::Load,
+                    store: true,
+                },
+            }],
+            depth_stencil_attachment: None,
+        };
+
+        let draw_functions = world
+            .get_resource::<DrawFunctions<TransparentUi>>()
+            .unwrap();
+
+        let render_pass = render_context
+            .command_encoder
+            .begin_render_pass(&pass_descriptor);
+
+        let mut draw_functions = draw_functions.write();
+        let mut tracked_pass = TrackedRenderPass::new(render_pass);
+        for item in transparent_phase.items.iter() {
+            let draw_function = draw_functions.get_mut(item.draw_function).unwrap();
+            draw_function.draw(world, &mut tracked_pass, view_entity, item);
+        }
+        Ok(())
+    }
+}
+
+pub struct TransparentUi {
+    pub sort_key: FloatOrd,
+    pub entity: Entity,
+    pub pipeline: CachedPipelineId,
+    pub draw_function: DrawFunctionId,
+}
+
+impl PhaseItem for TransparentUi {
+    type SortKey = FloatOrd;
+
+    #[inline]
+    fn sort_key(&self) -> Self::SortKey {
+        self.sort_key
+    }
+
+    #[inline]
+    fn draw_function(&self) -> DrawFunctionId {
+        self.draw_function
+    }
+}
+
+impl EntityPhaseItem for TransparentUi {
+    #[inline]
+    fn entity(&self) -> Entity {
+        self.entity
+    }
+}
+
+impl CachedPipelinePhaseItem for TransparentUi {
+    #[inline]
+    fn cached_pipeline(&self) -> CachedPipelineId {
+        self.pipeline
+    }
+}
+
+pub type DrawUi = (
+    SetItemPipeline,
+    SetUiViewBindGroup<0>,
+    SetUiTextureBindGroup<1>,
+    DrawUiNode,
+);
+
+pub struct SetUiViewBindGroup<const I: usize>;
+impl<const I: usize> EntityRenderCommand for SetUiViewBindGroup<I> {
+    type Param = (SRes<UiMeta>, SQuery<Read<ViewUniformOffset>>);
+
+    fn render<'w>(
+        view: Entity,
+        _item: Entity,
+        (ui_meta, view_query): SystemParamItem<'w, '_, Self::Param>,
+        pass: &mut TrackedRenderPass<'w>,
+    ) -> RenderCommandResult {
+        let view_uniform = view_query.get(view).unwrap(); // TODO: store bind group as component?
```

Comment:
The bind group is used globally (because we use a dynamic uniform). No need to store it on each individual view. We already store the offset as a per-view component, which feels right to me.